security/oc_pki.c

```diff
@@ -23,14 +23,17 @@
 #include "oc_pki.h"
 #include "oc_certs_internal.h"
 #include "oc_cred_internal.h"
+#include "oc_pki_internal.h"
 #include "oc_store.h"
 #include "oc_tls.h"
 #include "port/oc_connectivity.h"
 
+#include <mbedtls/ecp.h>
 #include <mbedtls/pk.h>
 #include <mbedtls/x509_crt.h>
 
 static oc_pki_verify_certificate_cb_t g_verify_certificate_cb;
+static const oc_pki_pk_functions_t *g_pk_functions;
```

Comment:
Storing a pointer is dangerous if somebody sends a pointer to local variables to `oc_pki_set_pk_functions`. 

I would prefer `oc_pki_pk_functions_t` + `bool` for safety.